File: backend/src/main/java/com/langleon/dsobuildsim/character/dto/CharacterDTO.java
package com.langleon.dsobuildsim.character.dto;

import com.langleon.dsobuildsim.buffs.dto.BuffInstanceDTO;
import com.langleon.dsobuildsim.character.CharacterClass;
import com.langleon.dsobuildsim.character.ClassSkillType;
import com.langleon.dsobuildsim.character.MasteryType;
import com.langleon.dsobuildsim.collectorbagbonus.dto.instance.CollectorBagCategoryBonusInstanceDTO;
import com.langleon.dsobuildsim.dragonstones.dto.DragonCrestTrinketDTO;
import com.langleon.dsobuildsim.essences.dto.EssenceInstanceDTO;
import com.langleon.dsobuildsim.items.core.enums.ItemSlot;
import com.langleon.dsobuildsim.items.dto.ItemInstanceDTO;
import com.langleon.dsobuildsim.jewels.dto.JewelTrinketDTO;
import com.langleon.dsobuildsim.pets.dto.PetInstanceDTO;
import com.langleon.dsobuildsim.runes.dto.RuneTrinketDTO;
import com.langleon.dsobuildsim.wisdomskilltree.dto.instance.WisdomSkillTreeInstanceDTO;

import java.util.List;
import java.util.Map;

public record CharacterDTO(CharacterClass characterClass,
                           String name,
                           MasteryType masteryType,
                           int masteryLevel,
                           ClassSkillType classSkillType,
                           int classSkillLevel,
                           List<RuneTrinketDTO> runeTrinkets,
                           List<JewelTrinketDTO> jewelTrinkets,
                           DragonCrestTrinketDTO dragonCrest,
                           Map<ItemSlot, ItemInstanceDTO> items,
                           PetInstanceDTO pet,
                           EssenceInstanceDTO essence,
                           BuffInstanceDTO tonic,
                           BuffInstanceDTO physic,
                           WisdomSkillTreeInstanceDTO wisdomSkillTree,
                           List<CollectorBagCategoryBonusInstanceDTO> collectorBagBuffs
                           ) {
}

File: frontend/src/app/models/instanceDTOs/CharacterDTO.ts
import { CharacterClass } from "../../enums/CharacterClass";
import { ClassSkillType } from "../../enums/ClassSkillType";
import { ItemSlot } from "../../enums/ItemSlot";
import { MasteryType } from "../../enums/MasteryType";
import { WisdomSkillType } from "../../enums/WisdomSkillType";
import { BuffInstanceDTO } from "./BuffInstanceDTO";
import { CollectorBagCategoryBonusInstanceDTO } from "./CollectorBagCategoryBonusInstanceDTO";
import { DragonCrestTrinketDTO } from "./DragonCrestTrinketDTO";
import { EssenceInstanceDTO } from "./EssenceInstanceDTO";
import { ItemInstanceDTO } from "./ItemInstanceDTO";
import { JewelTrinketDTO } from "./JewelTrinketDTO";
import { PetInstanceDTO } from "./PetInstanceDTO";
import { RuneTrinketDTO } from "./RuneTrinketDTO";
import { WisdomSkillInstanceDTO } from "./WisdomSkillInstanceDTO";

export interface CharacterDTO {
    characterClass: CharacterClass;
    name: string;
    masteryType: MasteryType;
    masteryLevel: number;
    classSkillType: ClassSkillType;
    classSkillLevel: number;
    runeTrinkets: RuneTrinketDTO[];
    jewelTrinkets: JewelTrinketDTO[];
    dragonCrest: DragonCrestTrinketDTO;
    items: Partial<Record<ItemSlot, ItemInstanceDTO>>;
    pet: PetInstanceDTO | null;
    essence: EssenceInstanceDTO | null;
    tonic: BuffInstanceDTO | null;
    physic: BuffInstanceDTO | null;
    wisdomSkills: Record<WisdomSkillType, WisdomSkillInstanceDTO>;
    collectorBagBuffs: CollectorBagCategoryBonusInstanceDTO[];
}